eos/scattering/ee-to-ccbar.cc

```diff
@@ -0,0 +1,778 @@
+                        channel_array[i] = std::make_shared<PWavePPChannel<EEToCCBar::nchannels, EEToCCBar::nresonances>>(channel_names[i], m_D, m_D, q[i], _get_g0_column(_filter_channel_index(Channels(i)), std::make_index_sequence<EEToCCBar::nresonances>()));
+                        break;
+
+                    default:
+                        throw InternalError("The number of requested channels (array) is larger than the number of known channel names.");
+                }
+            }
+
+            K = std::shared_ptr<KMatrix<EEToCCBar::nchannels, EEToCCBar::nresonances>> (
+                new KMatrix<EEToCCBar::nchannels, EEToCCBar::nresonances>(
+                    channel_array,
+                    resonance_array,
+                    _get_bkgcst_matrix(std::make_index_sequence<EEToCCBar::nchannels>(), std::make_index_sequence<EEToCCBar::nchannels>()),
+                    "e^+e^-->ccbar"
+                    )
+                );
+        }
+
+        const IntermediateResult * prepare(const complex<double> & E)
+        {
+            // Amplitude on the first RS
+            _intermediate_result.tmatrix_row_0 = K->tmatrix_row(0, E * E);
+            // Amplitude on the second RS
+            _intermediate_result.tmatrix2_row_0 = K->tmatrix_row(0, E * E, true);
+
+            _intermediate_result.E = E;
+            _intermediate_result.s = E * E;
+
+            return &_intermediate_result;
+        }
+
+
+        double rho(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            return std::real(K->_channels[Channels(channel)]->rho(intermediate_result->s));
+        }
+
+        complex<double> chew_mandelstam(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            return K->_channels[Channels(channel)]->chew_mandelstam(intermediate_result->s);
+        }
+
+        complex<double> chew_mandelstam_II(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            complex<double> s = intermediate_result->s;
+            auto get_channel = K->_channels[Channels(channel)];
+
+            const unsigned li = get_channel->_l_orbital;
+            const double q0 = get_channel->_q0.evaluate();
+            const complex<double> mi1_2 = power_of<2>((double)get_channel->_m1);
+            const complex<double> mi2_2 = power_of<2>((double)get_channel->_m2);
+
+            // Momentum of particles in their center-of-momentum frame
+            const complex<double> q = 0.5 * sqrt(eos::lambda(s, mi1_2, mi2_2)) / sqrt(s);
+
+            // Blatt-Weisskopf factors, cf eq. (50.26)
+            const complex<double> Fi = kmatrix_utils::blatt_weisskopf_factor(li, q / q0);
+
+            return get_channel->chew_mandelstam(s) +
+                complex<double>(0.0, 2.0) * get_channel->rho(s) * power_of<2>(pow(q / q0, li) * Fi);
+        }
+
+
+        inline double sigma_eetomumu(const double & E)
```

Comment:
Be careful with the naming; this is the LO theory expression for the cross section, correct? Can we append ``_theory`` or ``_leading_order`` or something to the name, to make clear that this is not an observable that we predict in a resonance model?